src/Analyzers/CSharp/Tests/RemoveUnusedMembers/RemoveUnusedMembersTests.cs

```diff
@@ -40,6 +40,23 @@ public void M()
             await VerifyCS.VerifyCodeFixAsync(code, code);
         }
 
+        [Fact, WorkItem(47917, "https://github.com/dotnet/roslyn/issues/47917")]
+        public async Task FieldIsNotReadButHaveDataMemberAttribute()
+        {
+            var code = @"
+using System.Runtime.Serialization;
+
+public class C
+{
+    [DataMember]
+    private string _x;
+
+    public C(string x) => _x = x;
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(code, code);
```

Comment:
For some reason this is failing:

```
Microsoft.CodeAnalysis.Testing.Verifiers.EqualWithMessageException : Context: Diagnostics of test state
Mismatch between number of diagnostics returned, expected "0" actual "3"

Diagnostics:
// /0/Test0.cs(6,6): error CS0246: The type or namespace name 'DataMember' could not be found (are you missing a using directive or an assembly reference?)
DiagnosticResult.CompilerError("CS0246").WithSpan(6, 6, 6, 16).WithArguments("DataMember"),
// /0/Test0.cs(6,6): error CS0246: The type or namespace name 'DataMemberAttribute' could not be found (are you missing a using directive or an assembly reference?)
DiagnosticResult.CompilerError("CS0246").WithSpan(6, 6, 6, 16).WithArguments("DataMemberAttribute"),
// /0/Test0.cs(7,20): info IDE0052: Private member 'C._x' can be removed as the value assigned to it is never read.
VerifyCS.Diagnostic(CSharpRemoveUnusedMembersDiagnosticAnalyzer.IDE0052).WithSpan(7, 20, 7, 22),\r\n\r\n\r\nAssert.Equal() Failure
Expected: 0
Actual:   3
```

The required `using` exists. I'm not sure why the type isn't found.

The test code compiles without errors in sharplab.